src/flows/pipes/Notification/Expressions.tsx

```diff
@@ -0,0 +1,195 @@
+import React, {FC, useState, useMemo, useCallback, useContext} from 'react'
+
+import {
+  Input,
+  InputType,
+  IconFont,
+  EmptyState,
+  ComponentSize,
+} from '@influxdata/clockface'
+import {FromFluxResult} from '@influxdata/giraffe'
+import Expression from 'src/flows/pipes/Notification/Expression'
+import {FlowContext} from 'src/flows/context/flow.current'
+
+interface Props {
+  parsed: FromFluxResult
+  id: string
+}
+
+interface SchemaExpressions {
+  [key: string]: Array<string>
+}
+
+const Expressions: FC<Props> = ({id, parsed}) => {
+  const [search, setSearch] = useState('')
+  const {flow, updateData} = useContext(FlowContext)
+  const updateSearch = useCallback(
+    e => {
+      setSearch(e.target.value)
+    },
+    [search, setSearch]
+  )
+
+  const parsedResultToSchema = (parsed: FromFluxResult): SchemaExpressions => {
+    let ni
+    const filtered = [
+      /^_start$/,
+      /^_stop$/,
+      /^_time$/,
+      /^_value/,
+      /^_measurement$/,
+      /^_field$/,
+      /^table$/,
+      /^result$/,
+    ]
+    if (!parsed) {
+      return
+    }
+
+    const out = parsed.table as any
+    const len = out.length
+    const measurements = out.columns._measurement?.data
+    const fields = out.columns._field?.data
+    const columns = out.columnKeys.filter(key => {
+      return filtered.reduce((acc, curr) => {
+        return acc && !curr.test(key)
+      }, true)
+    })
+    const schema = {
+      measurements: new Set<string>(),
+      fields: new Set<string>(),
+      tags: new Set<string>(),
+      columns: [],
+      system: [],
+    }
+
+    for (ni = 0; ni < len; ni++) {
```

Comment:
this might be faster... been a bit since i perf'd sets like this
```
new Set(measurements.filter(m => m.toLowerCase().includes(search.toLowerCase())))
new Set(fields.filter(f => f.toLowerCase().includes(search.toLowerCase())))
... maybe tags? ...
```

Reply:
nice yeah I was able to refactor that function and clean it up more with this suggestion